### PR TITLE
feat: platform context builder

### LINKED ISSUE #13046: Platform Context Builder
As we expand the number of things in the platform context we need better ways of configuring it. We use the builder pattern in a lot of other places, it makes sense to do so here as well.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #13046

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/BootstrapUtils.java [hunk 1]
```diff
@@ -89,20 +90,22 @@ public final class BootstrapUtils {
...
+     * @param settingsPath         the path to the settings.txt file (does not actually have to be named
+     *                             "settings.txt")
      */
-    public static void setupConfigBuilder(
-            @NonNull final ConfigurationBuilder configurationBuilder, @NonNull final Path settingsPath)
-            throws IOException {
-
-        final ConfigSource settingsConfigSource = LegacyFileConfigSource.ofSettingsFile(settingsPath);
-        final ConfigSource mappedSettingsConfigSource = ConfigMappings.addConfigMapping(settingsConfigSource);
+    public static void readSettingsDotTxt(
```
Comment (kfa-aguda) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/BootstrapUtils.java:99]: considering this doesn't have to be named settings.txt, will it be helpful to not name it readSettingsDotTxt? Maybe we name it readLegacySettings instead?

**Review by kfa-aguda — APPROVED** [review]
changes lgtm with just one nitpick.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/BootstrapUtils.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/BootstrapUtils.java:99]: Renamed to `readLegacySettingsFile()`

**Review by edward-swirldslabs — DISMISSED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM from `hedera-base` changes. Thanks @cody-littley

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformBuilder.java [hunk 2]
```diff
@@ -353,7 +352,6 @@ public PlatformComponentBuilder buildComponentBuilder() {
         }
 
         // FUTURE WORK: remove after static use of cryptography is abolished
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformBuilder.java:354]: The cryptography holder was removed but the comment about removing the cryptography holder was not.

**Review by edward-swirldslabs — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformBuilder.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformBuilder.java:354]: Comment is still relevant. There is also static reference to `MerkleCryptography` we still set here. I consider this to be part of getting rid of static cryptography.

Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformBuilder.java:354]: Comment removed. I also found I had to move the static instantiation of merkle cryptography to the context builder (issue exposed by failing JRS test).

**Review by edward-swirldslabs — APPROVED** [review]

**Comment by cody-littley** [comment]
Too many merge conflicts -- closing this branch and making change to clean branch.

**Pull request closed without merging**